openai/api_resources/file.py

```diff
@@ -27,35 +29,61 @@ def create(
         requestor = api_requestor.APIRequestor(
             api_key,
             api_base=api_base or openai.api_base,
+            api_type=api_type,
             api_version=api_version,
             organization=organization,
         )
-        url = cls.class_url()
+        typed_api_type, api_version = cls._get_api_type_and_version(api_type, api_version)
+
+        if typed_api_type == ApiType.AZURE:
+            base = cls.class_url()
+            url = "/%s%s?api-version=%s" % (cls.azure_api_prefix, base, api_version)
+        elif typed_api_type == ApiType.OPEN_AI:
+            url = cls.class_url()
+        else:
+            raise error.InvalidAPIType('Unsupported API type %s' % api_type)            
+
         # Set the filename on 'purpose' and 'model' to None so they are
         # interpreted as form data.
         files = [("purpose", (None, purpose))]
         if model is not None:
             files.append(("model", (None, model)))
         if user_provided_filename is not None:
-            files.append(("file", (user_provided_filename, file)))
+            files.append(("file", (user_provided_filename, file, 'application/octet-stream')))
         else:
-            files.append(("file", file))
+            files.append(("file", ("unknown", file, 'application/octet-stream')))
```

Comment:
@rachellim or @christinakim Could one of you confirm that changing this here and adding the name "unknown" is ok and won't impact anything? I'm not familiar with our file api and how it handles this.

Reply:
Tested this locally and took a quick look at our backend, I think this should be ok.

@sorinsuciu-msft Could we change "unknown" to "file"? I think for some reason it's the current behavior if no name is specified, so that should then keep it consistent.

Reply:
Sure, np